react/features/welcome/components/WelcomePage.native.js

```diff
@@ -14,56 +17,89 @@ import { ColorSchemeRegistry } from '../../base/color-scheme';
 import { translate } from '../../base/i18n';
 import { Icon, IconMenu, IconWarning } from '../../base/icons';
 import { MEDIA_TYPE } from '../../base/media';
-import { Header, LoadingIndicator, Text } from '../../base/react';
+import { LoadingIndicator, Text } from '../../base/react';
 import { connect } from '../../base/redux';
 import { ColorPalette } from '../../base/styles';
 import {
     createDesiredLocalTracks,
     destroyLocalDesktopTrackIfExists,
     destroyLocalTracks
 } from '../../base/tracks';
-import { HelpView } from '../../help';
-import { DialInSummary } from '../../invite';
-import { SettingsView } from '../../settings/components';
-import { setSideBarVisible } from '../actions';
+import BaseTheme from '../../base/ui/components/BaseTheme';
 
 import {
     AbstractWelcomePage,
-    _mapStateToProps as _abstractMapStateToProps
+    _mapStateToProps as _abstractMapStateToProps,
+    type Props as AbstractProps
 } from './AbstractWelcomePage';
 import LocalVideoTrackUnderlay from './LocalVideoTrackUnderlay';
 import VideoSwitch from './VideoSwitch';
 import WelcomePageLists from './WelcomePageLists';
-import WelcomePageSideBar from './WelcomePageSideBar';
 import styles, { PLACEHOLDER_TEXT_COLOR } from './styles';
 
+
+type Props = AbstractProps & {
+
+    /**
+     * The color schemed style of the Header component.
+     */
+    _headerStyles: Object,
+
+    /**
+     * Default prop for navigating between screen components(React Navigation).
+     */
+    navigation: Object,
+
+    /**
+     * Default prop for navigating between screen components(React Navigation).
+     */
+    route: Object,
+
+    /**
+     * The translate function.
+     */
+    t: Function
+};
+
 /**
  * The native container rendering the welcome page.
  *
  * @augments AbstractWelcomePage
  */
-class WelcomePage extends AbstractWelcomePage {
+class WelcomePage extends AbstractWelcomePage<*> {
     /**
      * Constructor of the Component.
      *
      * @inheritdoc
      */
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
 
+        // $FlowExpectedError
         this.state._fieldFocused = false;
+
+        // $FlowExpectedError
         this.state.hintBoxAnimation = new Animated.Value(0);
 
         // Bind event handlers so they are only bound once per instance.
         this._onFieldFocusChange = this._onFieldFocusChange.bind(this);
-        this._onShowSideBar = this._onShowSideBar.bind(this);
         this._renderHintBox = this._renderHintBox.bind(this);
 
         // Specially bind functions to avoid function definition on render.
         this._onFieldBlur = this._onFieldFocusChange.bind(this, false);
         this._onFieldFocus = this._onFieldFocusChange.bind(this, true);
     }
 
+    _onFieldBlur: () => void;
```

Comment:
are these necessary?

Reply:
These helped fix flow errors, that is why I added them.

Reply:
ok